bastion/gui/dashboard/main_window.py

```
@@ -0,0 +1,813 @@
+                    res = subprocess.run(['wc', '-l', str(self.log_path)], capture_output=True, text=True)
+                    total = res.stdout.strip().split()[0] if res.returncode == 0 else "0"
+                    self.stat_connections_label.setText(total)
+
+                    res = subprocess.run(['grep', '-c', 'decision: deny', str(self.log_path)],
+                                       capture_output=True, text=True)
+                    denied = res.stdout.strip() if res.returncode == 0 else "0"
+                    self.stat_blocked_label.setText(denied)
+                else:
+                    # Need elevated privileges
+                    cmd = ['pkexec', 'sh', '-c', f'wc -l "{self.log_path}" | cut -d" " -f1']
+                    res = subprocess.run(cmd, capture_output=True, text=True)
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)